pkg/middlewares/aws_defaults.go

```diff
@@ -0,0 +1,204 @@
+package middlewares
+
+import (
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+const defaultIamRolePathGlob = "/aws-service-role/*"
+
+// When scanning a brand new AWS account, some users may see irrelevant results about default AWS role policies.
+// We ignore these resources by default when strict mode is disabled.
+type AwsDefaults struct{}
+
+func NewAwsDefaults() AwsDefaults {
+	return AwsDefaults{}
+}
+
+func awsIamRoleDefaults(remoteResources, resourcesFromState *[]resource.Resource) error {
+	newRemoteResources := make([]resource.Resource, 0)
+
+	for _, remoteResource := range *remoteResources {
+		// Ignore all resources other than iam role
+		if remoteResource.TerraformType() != aws.AwsIamRoleResourceType {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		existInState := false
+		for _, stateResource := range *resourcesFromState {
```

Comment:
I don't think we should check if the resource is managed ? @wbeuil Should we ignore theses cloud provider defaults only if they are not managed ? I'm not sure this makes sense to manage theses kind of resources using terraform (imported in terraform), some of them cannot be modified for example.

Reply:
I've never managed the default with terraform so maybe @sjourdan can help answering this question. But as for the "should we ignore the defaults only if they are not managed`, I answer let's ignore them all the time.

Reply:
Shouldn't we consider users may want to manage those resources anyway ? If we ignore them all the time and the user manages them, the resources will appear as missing on cloud provider. Not sure how we should handle that behavior then.

Reply:
We can exclude them from `resourcesFromState` too

Reply:
Right, and the user can use the strict flag when they want to manage these resources. It makes sense

Reply:
Exactly

Reply:
up